### ISSUE TITLE
Add downloads tab (ability to sort and watch offline videos)

### ISSUE THREAD

**Issue by peepo5** [FreeTubeApp/FreeTube#1772]
**Is your feature request related to a usage problem (not a bug)? Please describe.**
As of now, downloading videos in FreeTube is finnicky, slightly unreliable and not manageable / accessible in the app. There are many problems with the current implimentation so here is a list of the main problems:
1. Confusing download options (and sorted weirdly)
2. Not a direct download link (directs to browser and you also need to right click to download)
3. Does not download video information such as thumbnail, title or description
4. Downloaded videos are not viewable in the app
5. Download playlists / channels and view inside the app (good luck)

**Describe the solution you'd like to see implemented**
There would be many different additions needed to add a downloads tab. The first would be making downloads go to a local directory rather than just opening a link in a browser. The second would be mapping video information to video files without making the folder structure look ugly (basically meaning downloaded files can be moved / copied easily). The third would be displaying the offline videos and video information inside of the app, which would require a very different implementation than it is currently.

TLDR; I would like for a downloads tab to be added.

(As of writing) There is a bug where when you are offline, a blank white theme is displayed and the user cannot access even the settings. This needs to be fixed to start implementing a tab.

**Describe alternatives you've considered**
Viewing videos manually outside freetube

**Additional context**
This would help people who are using laptops / mobile devices and it would also help people that want to archive videos locally. It would make it alot more efficient to download videos and more manageable.

For now, I think the focus should be on making videos download directly to a directory before any of the other suggestions are looked at.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Duplicates of https://github.com/FreeTubeApp/FreeTube/issues/582, https://github.com/FreeTubeApp/FreeTube/issues/643, https://github.com/FreeTubeApp/FreeTube/issues/884, #1434

**Comment by peepopoggers** [comment]
@efb4f5ff-1298-471a-8973-3d47447115dc
This is not a duplicate, none of the issues describe adding a downloads tab, only improving downloads in their current form. If you think this PR is unnecessary then talk in matrix.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Oh i see. Apologies for the mistake.

**Comment by constraintAutomaton** [comment]
I would be very interested in tacking this issue, but It might be good to divide it into part. I would like to start by creating a page showing an history of download and reading the video in app.

**Comment by constraintAutomaton** [comment]
I would like to tackle it starting by a download page and in app download

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Im looking forward to see ur PR😄

**Comment by chrrybmb64** [comment]
Suggestions for a downloads tab:

-   Show all of the video(s) which you are currently downloading
-   Show the progress of your downloads in the form of a progress bar or percentage (or both)
-   Show you information about your downloads such as file size, video format, resolution, framerate, and download path
-   Show you videos which you've downloaded in the past
-   Let you pause, resume, and cancel downloads
-   Let you queue videos to be downloaded one at a time

**Comment by TheodorSmall** [comment]
Some ideas that can make downloads an even more powerful feature, however the time to implement these might not be worth it.
First, "local" videos could be introduced, to which downloaded videos can be saved, including the following data:
  - video ID (if it is not available, because the video wasn't fetched from YT or modified afterwards, a local ID in another format can be created)
  - metadata such as title, description, chapters, thumbnail
  - the video itself in one or more formats
  - according to a setting, screenshots can be saved with the video (even if the video itself hasn't been downloaded)
Next, these local videos could also be added from the local filesystem, or parts of it can be changed by the user (for example, there are lots of videos, especially music compilations, that don't use the chapters feature, but timestamps are given in the comments or the description). Of course, external tools can be used here, which would suggest some flag telling the app to currently ignore this video.
By saving the YT video ID (I don't know what it's called officially), an indicator can tell the user whether any displayed video (or format) is locally available.
These local videos can then be treated like any other "normal" video, e.g. added to playlists, or optionally be displayed when searching, in an extra section above the results returned by YT.

**Comment by Pentaphon** [comment]
I agree. There is no download UI at all. No progress bar, no download history, nothing. There needs to be a tab and a progress bar.
